examples/births-deaths/population_manager.rs

```diff
@@ -0,0 +1,353 @@
+}
+
+impl ContextPopulationExt for Context {
+    fn attempt_death(&mut self, person_id: PersonId) {
+        self.set_person_property(person_id, Alive, false);
+    }
+
+    fn create_new_person(&mut self, birth_time: f64) -> PersonId {
+        let person = self.add_person();
+        self.initialize_person_property(person, Birth, birth_time);
+        self.initialize_person_property(person, Alive, true);
+        person
+    }
+
+    fn get_person_age_group(&mut self, person_id: PersonId) -> AgeGroupRisk {
+        let current_age = self.get_person_age(person_id);
+        if current_age <= 1.0 {
+            AgeGroupRisk::NewBorn
+        } else if current_age <= 65.0 {
+            AgeGroupRisk::General
+        } else {
+            AgeGroupRisk::OldAdult
+        }
+    }
+    fn get_person_age(&mut self, person_id: PersonId) -> f64 {
+        let birth_time = self.get_person_property(person_id, Birth);
+        (self.get_current_time() - birth_time) / 365.0
+    }
+    fn get_current_group_population(&mut self, age_group: AgeGroupRisk) -> usize {
+        let mut current_population = 0;
+        for i in 0..self.get_current_population() {
+            let person_id = self.get_person_id(i);
+            if self.get_person_property(person_id, Alive)
+                && self.get_person_age_group(person_id) == age_group
+            {
+                current_population += 1;
+            }
+        }
+        current_population
+    }
+
+    fn sample_person(&mut self, age_group: AgeGroupRisk) -> Option<PersonId> {
+        let mut people_vec = Vec::<PersonId>::new();
+        for i in 0..self.get_current_population() {
+            let person_id = self.get_person_id(i);
+            if self.get_person_property(person_id, Alive)
+                && self.get_person_age_group(person_id) == age_group
+            {
+                people_vec.push(person_id);
+            }
+        }
+        if people_vec.is_empty() {
+            None
+        } else {
+            Some(people_vec[self.sample_range(PeopleRng, 0..people_vec.len())])
+        }
+    }
+    fn get_population_by_property<T: PersonProperty + 'static>(
+        &mut self,
+        property: T,
+        value: T::Value,
+    ) -> usize
+    where
+        <T as PersonProperty>::Value: PartialEq,
+    {
+        let mut population_counter = 0;
+        for i in 0..self.get_current_population() {
+            let person_id = self.get_person_id(i);
+            if self.get_person_property(person_id, property) == value {
+                population_counter += 1;
+            }
+        }
+        population_counter
+    }
+
+    fn sample_person_by_property<T: PersonProperty + 'static>(
+        &mut self,
+        property: T,
+        value: T::Value,
+    ) -> Option<PersonId>
+    where
+        <T as PersonProperty>::Value: PartialEq,
+    {
+        let mut people_vec = Vec::<PersonId>::new();
+        for i in 0..self.get_current_population() {
```

Comment:
I would probably use `filter_map` here but this is perfectly fine

Reply:
Noted. I'm not sure I should work too much on this sample since it will be replaced once we land the PR to query people, right?